superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -124,6 +205,145 @@ export default class ResultSet extends React.PureComponent<
     }
   }
 
+  handleOnChangeAutoComplete = () => {
+    this.setState({ datasetToOverwrite: {} });
+  };
+
+  handleOverwriteDataset = async () => {
```

Comment:
`async/await` usually used as a pair, to avoid nested `callback hells`. Could you please replace `.then` block with await?